crates/task-impls/src/view_sync.rs

```diff
@@ -277,25 +278,22 @@ impl<
             HotShotEvent::ViewSyncPreCommitVoteRecv(ref vote) => {
                 let mut map = self.pre_commit_relay_map.write().await;
                 let vote_view = vote.get_view_number();
-                if let Some(relay_task) = map.remove(&vote_view) {
+                let relay = vote.get_data().relay;
+                let relay_map = map.entry(vote_view).or_insert(BTreeMap::new());
+                if let Some(relay_task) = relay_map.remove(&relay) {
                     debug!("Forwarding message");
                     let result = relay_task.handle_event(event.clone()).await;
 
                     if result.0 == Some(HotShotTaskCompleted::ShutDown) {
                         // The protocol has finished
                         return;
                     }
-
-                    map.insert(vote_view, result.1);
+                    relay_map.insert(relay, result.1);
```

Comment:
Should we also remove the `vote_view` from the `map` if the `relay_map` is empty? We did that in the previous code, but now it looks like we only clear the inner `relay_map`.

Reply:
I think if remove the whole view, we'd just end up creating it again on the next vote from this view.  We still do the Garbage collection on view change to remove views from the map

Reply:
Makes sense!